### PR TITLE
[W06H02] remove/fix tests expecting specific values in unused mem array

### LINKED ISSUE #119: W06H02 pop test
```
void popShouldReturnCorrectValues() {
      final var value = 10;
      final var value2 = 20;
      final var value3 = 30;
      final var value4 = 40;

      stack.push(value);
      assertEquals(value, stack.pop());

      stack.push(value2);
      stack.push(value3);
      assertEquals(value3, stack.pop());

      assertArrayEquals(array(value2), getMem(getHead()));

      stack.push(value4);
      stack.push(value3);
      assertEquals(value3, stack.pop());
      assertEquals(value4, stack.pop());
      assertEquals(value2, stack.pop());

      assertArrayEquals(array(0), getMem(getHead()));
}
```

Why do we need the last `assertArrayEquals` ?

### LINKED ISSUE #125: [W06H02] popShouldReturnCorrectValues tests irrelevant Stack contents
`popShouldReturnCorrectValues()` prüft den Stack auf Gleichheit mit dem Array `int[]{0}`, nachdem das letzte verbleibende Wert aus dem Stack ge`pop()`ed wurde:
https://github.com/MaximilianAnzinger/pgdp2223-tests/blob/f925297e3696b4a5226aad552e08ad429638044d/w06h02/test/pgdp/ds/MultiStackTest.java#L231
Laut Tutoren ist der Inhalt der ge`pop()`ed Zellen aber egal:
[https://zulip.in.tum.de/PGdP-W06H02/pop](https://zulip.in.tum.de/#narrow/stream/1480-PGdP-W06H02/topic/pop.28.29.20muss.20mem.20ver.C3.A4ndert.20werden.3F)
Stattdessen müsste geprüft werden, ob der `top`-Pointer gleich `-1` ist (und evtl, dass nach dem Ersten Stack keine weiteren mehr sind):
```java
int getTopPointer(Stack s) {
    try {
        final var field = s.getClass().getDeclaredField("top");
        field.setAccessible(true);
        return (int) field.get(s);
    catch (IllegalAccessException | NoSuchFieldException e) {
        throw new RuntimeException(e);
    }
}
```
und in `popShouldReturnCorrectValues()`:
```java
assertEquals(-1, getTopPointer(getHead()));
```

### PR DESCRIPTION
**Pull request by felixhauptmann**
The task description does not specify whether or not the mem array must get modified by `pop()` ([confirmed](https://zulip.in.tum.de/#narrow/stream/1480-PGdP-W06H02/topic/pop.28.29.20muss.20mem.20ver.C3.A4ndert.20werden.3F)).

This pr removes/fixes tests which assume that unused memory has to be filled with zeros. Closes #119 and closes #125

### CODE REVIEW

**Comment by nilsreichardt** [comment]
@julian-hartl Can you review this change because it changes one of your tests

**Comment by tdphucvn** [comment]
LGTM

File: w06h02/test/pgdp/ds/MultiStackTest.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ void pushOutOfCapacityShouldCreateNewStack() {
         stack.push(value);
         final var next = getNext(getHead());
         final var mem = getMem(next);
-        assertTrue(arrayContains(mem, value));
+        assertEquals(mem[0], value);
```
Comment (julian-hartl) [inline w06h02/test/pgdp/ds/MultiStackTest.java:79]: Where does the exercise say that values have to be added to the front of the mem array? A valid solution would also be to add new values to the end of the array and make the stack pointer point there.

File: w06h02/test/pgdp/ds/MultiStackTest.java [hunk 2]
```diff
@@ -202,7 +202,7 @@ void topShouldReturnCorrectValues() {
 
         assertArrayEquals(array(value), getMem(getHead()));
         assertArrayEquals(array(value2, value3), getMem(getNext(getHead())));
-        assertArrayEquals(array(value4, 0, 0, 0), getMem(getNext(getNext(getHead()))));
+        assertEquals(value4, getMem(getNext(getNext(getHead())))[0]);
```
Comment (julian-hartl) [inline w06h02/test/pgdp/ds/MultiStackTest.java:205]: Same here and by the way: could you maybe extract this to a new variable for readability purposes?

File: w06h02/test/pgdp/ds/MultiStackTest.java [hunk 3]
```diff
@@ -227,7 +227,5 @@ void popShouldReturnCorrectValues() {
         assertEquals(value3, stack.pop());
         assertEquals(value4, stack.pop());
         assertEquals(value2, stack.pop());
-
-        assertArrayEquals(array(0), getMem(getHead()));
```
Comment (ipadmini22) [inline w06h02/test/pgdp/ds/MultiStackTest.java:231]: Would it make sense to add another of the Min_Value tests here, e.g. validating that after usage the stack has correctly been identified as empty? That would be a better replacement for the current tests, wouldn't it?

File: w06h02/test/pgdp/ds/MultiStackTest.java (hunk 1 above)
Comment (felixhauptmann) [reply w06h02/test/pgdp/ds/MultiStackTest.java:79]: Yeah you're right

File: w06h02/test/pgdp/ds/MultiStackTest.java (hunk 3 above)
Comment (felixhauptmann) [reply w06h02/test/pgdp/ds/MultiStackTest.java:231]: Yes that's a good idea. So we could just check if top/pop returns `Integer.MIN_VALUE`.

File: w06h02/test/pgdp/ds/MultiStackTest.java (hunk 2 above)
Comment (felixhauptmann) [reply w06h02/test/pgdp/ds/MultiStackTest.java:205]: 👍

File: w06h02/test/pgdp/ds/MultiStackTest.java (hunk 1 above)
Comment (felixhauptmann) [reply w06h02/test/pgdp/ds/MultiStackTest.java:79]: I just thought about implementing that... I wont do it. feel free to do it yourself but keep in mind that that you would have to change all other `mem` tests where `mem.length < 1` as well.

**Review by MaximilianAnzinger — APPROVED** [review]
[🐧](https://emojipedia.org/penguin/)

**Comment by felixhauptmann** [comment]
I think my recent change (proposed by @ipadmini22) makes sense. As I have said I wont implement checking if values are somewhere in `mem` or the end. Either one of you implements that (not just partially but for all assertions with `mem.length < 1`) or if someone really has a solution with a stack starting at `mem.length - 1` that's their problem.

**Comment by felixhauptmann** [comment]
This closes #119

**Comment by nilsreichardt** [comment]
@felixhauptmann Tag me, if this PR is ready to merge 👍

File: w06h02/test/pgdp/ds/MultiStackTest.java (hunk 1 above)
Comment (julian-hartl) [reply w06h02/test/pgdp/ds/MultiStackTest.java:79]: That's fine. Most people won't implement it that way anyways.

**Comment by felixhauptmann** [comment]
@nilsreichardt ready

**Review by MaximilianAnzinger — COMMENTED** [review]
[🐧](https://emojipedia.org/penguin/)

**Pull request merged**